rustls/src/server/builder.rs

```diff
@@ -64,6 +64,8 @@ impl ConfigBuilder<ServerConfig, WantsServerCert> {
     /// but other `CryptoProviders` may not.
     ///
     /// This function fails if `key_der` is invalid.
+    ///
+    /// This function fails if `key_der` is found to be inconsistent with `cert_chain`.
```

Comment:
I don't love this documentation addition. For one thing, IMO it would be better not to repeat the first part of the sentence, and we could instead concatenate them together somehow. Second, without context, I think "found to be inconsistent" is pretty vague. Proposal:

"This function fails if `key_der` is invalid, or if the subject public key info from the private key does not match the public key for the end entity certificate from the `cert_chain`."

Reply:
Done!